editor/src/main/java/io/github/rosemoe/sora/widget/SymbolInputView.java

```diff
@@ -122,8 +122,12 @@ public void addSymbols(@NonNull String[] display, @NonNull final String[] insert
             int finalI = i;
             btn.setOnClickListener((view) -> {
                 if (editor != null && editor.isEditable()) {
-                    if ("\t".equals(insertText[finalI]) && editor.getSnippetController().isInSnippet()) {
-                        editor.getSnippetController().shiftToNextTabStop();
+                    if ("\t".equals(insertText[finalI])) {
+                        if (editor.getSnippetController().isInSnippet()) {
+                            editor.getSnippetController().shiftToNextTabStop();
+                        } else if (editor.getProps().indentSelectionWithTab) {
+                            editor.indentSelection();
+                        }
```

Comment:
Missing fallback case for inserting the TAB.

Reply:
Sure, will fix this.